include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -336,6 +339,152 @@ class gap_decorator
+        using reference = typename gap_decorator::const_reference; // = reference
+        //!\brief Equals reference type.
+        using const_reference = reference;
+        //!\brief Pointer of container value type.
+        using pointer = value_type *;
+        //!\brief Signals that this is a random access iterator.
+        using iterator_category = std::random_access_iterator_tag;
+        //!\}
+
+        /*!\name Constructors/Destructors
+         * \{
+         */
+        constexpr gap_decorator_iterator_ra()                                              = default; //!< Defaulted.
+        constexpr gap_decorator_iterator_ra(gap_decorator_iterator_ra const &)             = default; //!< Defaulted.
+        constexpr gap_decorator_iterator_ra & operator=(gap_decorator_iterator_ra const &) = default; //!< Defaulted.
+        constexpr gap_decorator_iterator_ra (gap_decorator_iterator_ra &&)                 = default; //!< Defaulted.
+        constexpr gap_decorator_iterator_ra & operator=(gap_decorator_iterator_ra &&)      = default; //!< Defaulted.
+        ~gap_decorator_iterator_ra()                                                       = default; //!< Defaulted.
+
+        //!\brief Construct from seqan3::gap_decorator and initialise members.
+        explicit constexpr gap_decorator_iterator_ra(gap_decorator const & host_) noexcept : base_t{host_}
+        {}
+
+        //!\brief Construct from seqan3::gap_decorator and explicit position.
+        constexpr gap_decorator_iterator_ra(gap_decorator const & host_,
+                                            typename gap_decorator::size_type const pos_) noexcept : base_t{host_, pos_}
+        {}
+        //!\}
+
+        /*!\name Arithmetic operators
+        * \{
+        */
+        //!\brief Pre-increment, returns updated iterator.
+        constexpr gap_decorator_iterator_ra & operator++() noexcept
+        {
+            base_t::operator++();
+            return *this;
+        }
+
+        //!\brief Pre-decrement, returns updated iterator.
+        constexpr gap_decorator_iterator_ra & operator--() noexcept
+        {
+            base_t::operator--();
+            return *this;
+        }
+
+        //!\brief Post-increment, returns previous iterator state (delegates to pre-increment).
+        constexpr gap_decorator_iterator_ra operator++(int) noexcept
+        {
+            gap_decorator_iterator_ra cpy{*this};
+            ++(*this);
+            return cpy;
+        }
+
+        //!\brief Post-decrement, returns previous iterator state (delegates to pre-decrement).
+        constexpr gap_decorator_iterator_ra operator--(int) noexcept
+        {
+            gap_decorator_iterator_ra cpy{*this};
+            --(*this);
+            return cpy;
+        }
+
+        //!\brief Forward this iterator.
+        constexpr gap_decorator_iterator_ra & operator+=(difference_type const skip) noexcept
+        {
+            this->jump(this->pos + skip);
```

Comment:
this means, you always start from scratch looking for the new position right? So always a binary search. 
It might be more efficient to check first if you would stay in the same node and conduct only binary search if this is not the case (so for small jumps it would be as efficient as it can be). But I am not familiar with the data structures used and if you can access the size of the current node (but I guess it should)

Reply:
yes you are right but this is not trivial and would require some thought. I will add this to the todos and probably tackle it when revising the insert_gap function (to be more efficient with the iterator hint) in a separate PR?